crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -594,7 +596,11 @@ impl EulerBasis {
 #[pymethods]
 impl EulerBasis {
     fn __reduce__(&self, py: Python) -> Py<PyAny> {
-        (py.get_type::<Self>(), (PyString::new(py, self.as_str()),)).into_py(py)
+        (
+            py.get_type_bound::<Self>(),
+            (PyString::new_bound(py, self.as_str()),),
+        )
+            .into_py(py)
     }
```

Comment:
Looking again at this (I only just wrote it recently), this actually ought to be a `__getnewargs__` impl rather than a `__reduce__`, but that's not important for this PR.